internal/edge/poll.go

```diff
@@ -271,7 +274,7 @@ func (service *PollService) poll() error {
 		}
 	}
 
-	if responseData.Status == "REQUIRED" && !service.tunnelClient.IsTunnelOpen() {
+	if responseData.Status == "REQUIRED" && !service.tunnelClient.IsTunnelOpen() && service.tunnel {
```

Comment:
service.tunnel will never change during the running of the agent, so at least, the `service.tunnel` should come first - so the other 2 conditions are never executed.
Personally, service.tunnelClient should be nil in this case, and quite a bit of code should never run.

Reply:
agree. fixed.